src/soc/intel/common/block/graphics/graphics.c

```diff
@@ -278,6 +278,13 @@ static void graphics_dev_read_resources(struct device *dev)
 		pci_dev_set_resources(dev);
 		res_bar0->flags |= IORESOURCE_FIXED;
 	}
+
+	if (ENV_X86_32) {
+		/* Place framebuffer below 4G to allow drawing bootsplash */
+		struct resource *res_bar2 = find_resource(dev, PCI_BASE_ADDRESS_2);
+		res_bar2->limit = 0xffffffff;
+		res_bar2->flags &= ~(IORESOURCE_PCI64 | IORESOURCE_ABOVE_4G);
```

Comment:
```suggestion
		res_bar2->flags &= ~IORESOURCE_ABOVE_4G;
```

Should be enough.  The BAR is still 64bit and we should keep that info. The resource limit set to UINT32_MAX  (0xffffffff) should make coreboot  allocate it under 4G. Please try it and apply in other places too.

Reply:
Applied: https://github.com/Dasharo/coreboot/pull/609/commits/b890d38d0f14946e5f9c26631b6c64ac58e7d2ba